src/image/hfe.c

```diff
@@ -171,6 +171,9 @@ static void hfe_setup_track(
     } else {
         /* Write mode. */
         im->hfe.trk_pos = im->cur_bc / 8;
+        /* Provide context to the write to avoid corrupting an opcode. */
+        if (im->hfe.is_v3 && im->hfe.trk_pos > 2 && (im->hfe.trk_pos & 255) < 2)
```

Comment:
This jogs the start pos by up to two bytes, and doesn't handle `trk_pos == 1`.

More complex, but why not jog back `-1` if `trk_pos != 0 && trk_pos&255==0`, and jog forward `+1 % trk_len` if `trk_pos&255==1` ?

Reply:
I'd also accept jogging back `-1 % trk_len` if `trk_pos&255==0`

Reply:
Ah, yeah, I didn't handle trk_pos == 1. I think I'll just set it to trk_pos = 0.

Jogging forward at all wasn't worth it, as then I needed to account for running off the end of the track.

Reply:
Yeah I guess. Jog forward +1 would just need a test if `trk_pos==trk_len` and set to `0`.

Reply:
> I'd also accept jogging back -1 % trk_len if trk_pos&255==0

I was swapping to this approach and then realized it'd be busted for `trk_len & 255 == 1`. So I just special case the very beginning of the track.